src/dsp_tools/commands/xmlupload/check_consistency_with_ontology.py

```diff
@@ -0,0 +1,123 @@
+# sourcery skip: use-fstring-for-concatenation
+import regex
+from lxml import etree
+from regex import Pattern
+
+from dsp_tools.commands.xmlupload.models.ontology_diagnose_models import InvalidOntologyElements, OntoCheckInformation
+from dsp_tools.commands.xmlupload.ontology_client import OntologyClientLive
+from dsp_tools.models.exceptions import BaseError
+
+defaultOntologyColon: Pattern[str] = regex.compile(r"^:\w+$")
+knoraUndeclared: Pattern[str] = regex.compile(r"^\w+$")
+genericPrefixedOntology: Pattern[str] = regex.compile(r"^[A-Za-z]+-?[A-Za-z]+:\w+$")
+
+
+def do_xml_consistency_check(onto_client: OntologyClientLive, root: etree._Element) -> None:
+    """
+    This function takes an OntologyClient and the root of an XML.
+    It retrieves the ontologies from the server.
+    It iterates over the root.
+    If it finds any invalid properties or classes, they are printed out and a UserError is raised.
+
+     Args:
+         onto_client: client for the ontology retrieval
+         root: root of the XML
+
+     Raises:
+         UserError: if there are any invalid properties or classes
+    """
+    onto_check_info = OntoCheckInformation(
+        default_ontology_prefix=onto_client.default_ontology,
+        onto_lookup=onto_client.get_all_ontologies_from_server(),
+        save_location=onto_client.save_location,
+    )
+    classes, properties = _get_all_classes_and_properties(root)
+    _find_problems_in_classes_and_properties(classes, properties, onto_check_info)
+
+
+def _get_all_classes_and_properties(root: etree._Element) -> tuple[list[list[str]], list[list[str]]]:
+    classes = _get_all_class_types_and_ids(root)
+    properties = []
+    for resource in root.iterchildren(tag="resource"):
+        properties.extend(_get_all_property_names_and_resource_ids(resource))
+    return classes, properties
+
+
+def _get_all_class_types_and_ids(root: etree._Element) -> list[list[str]]:
+    return [[resource.attrib["id"], resource.attrib["restype"]] for resource in root.iterchildren(tag="resource")]
+
+
+def _get_all_property_names_and_resource_ids(resource: etree._Element) -> list[list[str]]:
```

Comment:
same here: the return values are the other way round: `_get_all_resource_ids_and_property_names`